### PR TITLE
Add i18n functionality

### PR DESCRIPTION
**Pull request by Arsfiqball**
- [x] FAQ
- [x] Projects

Now, we can add translation in:
``data/locale/<lang>/faq.json``
``data/locale/<lang>/faq/<filename.md>``
``data/locale/<lang>/projects.json``
``data/locale/<lang>/projects/<filename.md>``

At bottom left, you will see a select button to select language.
It will set variable ``lang`` to localStorage.
then, projects directive and faq directive will process it to translate content.

### CODE REVIEW

**Comment by sils** [comment]
Comment on 1df031ecfa0a24bd4d340368c26ccfa5d7f6bb68, file resources/js/app.js.

This file had 219 lines, which is 19 lines more than the maximum limit specified.

*LineCountBear, severity NORMAL, section `linecount`.*

File: data/locale/id/projects.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "roberta_external_source.md": {
+    "name" : "Linting dengan satu klik untuk proyek online di browser",
+    "desc" : "Buat sebuah loader untuk proyek-proyek dari sumber data remote yang bisa digunakan dari halaman web coala untuk menyediakan linting dalam satu klik untuk proyek-proyek online.",
+    "requirements" : [
+      "Pelamar (applicant) sebaiknya terbiasa dengan pembuatan bear dan XML parsing menggunakan xpath atau bahasa query yang sejenis",
+      "Pelamar sebaiknya terbiasa dengan codebase coala dan cara kerja dari REST APInya, termasuk otentikasi.",
+      "Pelamar telah berkontribusi ke modul coala yang relevan, platform Open Roberta, atau keduanya."
+    ],
+    "difficulty" : "high",
```
Comment (jayvdb) [inline data/locale/id/projects.json]: difficulty 'high' isnt translated ;)

File: data/locale/id/projects.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "issues" : [
+    ],
+    "mentors" : ["maweki"],
+    "initiatives" : ["GSoC"],
+    "tags" : ["Bears", "Web"],
+    "collaborating_projects" : ["coala", "openroberta"]
```
Comment (jayvdb) [inline data/locale/id/projects.json]: Several fields are not translatable, like this one and mentors. If they are not translatable, they shouldnt be duplicated in the locale file.

Comment (Arsfiqball) [inline data/locale/id/projects.json]: oh yeah, apologize. it just a quick copy paste. :smile:

File: .coafile [hunk 3]
```diff
@@ -7,7 +7,7 @@ ignore = vendors/**
 files = **.yml, **.html, **.md, **.js
 bears = LineCountBear
 ignore = vendors/**, data/projects.js
-max_lines_per_file = 200
+max_lines_per_file = 400
```
Comment (jayvdb) [inline .coafile]: no longer needed after ad51afed

File: resources/js/app.js [hunk 4]
```diff
@@ -113,20 +139,40 @@
                     }
                 }
 
-                $scope.projects_url_dict = {}
-                $scope.projects_url_list = Object.keys($scope.projects_url_dict);
-                angular.forEach($scope.projectList, function(value, key){
-                    value["url"] = encodeURIComponent(value["name"].split(' ').join('_').toLowerCase());
-                    $scope.projects_url_dict[value["url"]] = key
-                });
+                function projectRequest() {
```
Comment (hemangsk) [inline resources/js/app.js]: Using it as $scope.projectRequest() will be better

File: resources/js/app.js [hunk 5]
```diff
@@ -137,13 +183,46 @@
...
+    app.directive('faq',[ '$http', '$templateCache', function ($http, $templateCache) {
         return {
             restrict: 'E',
             templateUrl: '/partials/tabs/faq.html',
             controller: function () {
                 self = this
+                lang = window.localStorage.getItem('lang')
                 self.faqs = faq
+
+                function setFromDefault(key) {
```
Comment (hemangsk) [inline resources/js/app.js]: here as well, if you feel its only a part of this controller an won't reuse this function, self.setFromDefault will be better. If a reuse is a possibility, then I suggest using Factory

File: resources/js/app.js (hunk 4 above)
Comment (Arsfiqball) [inline resources/js/app.js]: That's what I thought :+1:

File: resources/js/app.js (hunk 5 above)
Comment (Arsfiqball) [inline resources/js/app.js]: we can improve reusability later... :)

**Comment by jayvdb** [comment]
ack fd5e454

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**